### PR TITLE
Scala3 migration guide

### PR DESCRIPTION
**Pull request by PeuTit**
#11716

I think it's a good idea to document all the things we stumpled into during our own journey.

This is WIP.

TODO:

- [ ] Go through the https://github.com/playframework/playframework/labels/scala3 label and see what else should be mentioned here.
- [ ] Come up with better wording for `onLoadMessage` in `PlayAkkaHttpServer` related to Scala 3, see https://github.com/playframework/playframework/pull/11738#issuecomment-1490238354
- [ ] Mention 
```
libraryDependencies += clusterSharding
libraryDependencies += ("com.typesafe.akka" %% "akka-cluster-sharding-typed" % "2.6.20").cross(CrossVersion.for3Use2_13)
excludeDependencies += sbt.ExclusionRule("com.typesafe.akka", "akka-cluster-sharding-typed_3")
```
because it is not possible to override / replace the `clusterSharding` dependency from the akka-http sbt plugin.

### CODE REVIEW

File: documentation/manual/Scala3Migration.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+In Scala 3, you will need to update it to the following:
+
+```scala
+Form(
+	mapping(
+		"title" -> nonEmptyText,
+		"body" -> text
+	)(PostFormInput.apply)(t => Some(t.title, t.body))
```
Comment (xuwei-k) [inline documentation/manual/Scala3Migration.md:68]:
```suggestion
	)(PostFormInput.apply)(t => Some(Tuple.fromProductTyped(t)))
```

We can use `fromProductTyped` if unnecessary cross build with Scala 2.13.

- https://github.com/lampepfl/dotty/issues/2335
- https://github.com/lampepfl/dotty/blob/3.3.0-RC3/library/src/scala/Tuple.scala#L256

Comment (PeuTit) [reply documentation/manual/Scala3Migration.md]: This does not compile for me in Scala 2.13 but it does in 3.3.0-RC3. The error I get is:
`not found: value Tuple`

Tested with the following code:
```scala
    Form(
      mapping(
        "title" -> nonEmptyText,
        "body" -> text
      )(PostFormInput.apply)(t => Some(Tuple.fromProductTyped(t)))
    )
```

**Comment by mkurz** [comment]
Need to document https://github.com/playframework/playframework/issues/11846

**Comment by mkurz** [comment]
Need to mention #11883

**Comment by mkurz** [comment]
Incorporated into #11804

**Pull request closed without merging**